### ISSUE TITLE
Frost Strike improper scaling

### ISSUE THREAD

**Issue by taralej90** [dalaranwow/dalaran-wow#2390]
I really need the community's help on this one and I'd rather open an issue here and use it as a researching discussion than a thread on the forum, as nobody will read it there (seriously, the forum is pretty bad and abandoned).

I noticed that there are a lot of 80lvl Frost DKs raiding on Algalon before I even maxed my first 80. Needless to say, this shouldn't be the case as Unholy is quite strong on this content, if not even stronger than Frost. Well, at least it should be.

I've done a lot of testing with a friend's Unholy DK and everything seems to work OK in that tree; however, things are not like that for Frost. Many people say that Frost is bugged - Killing Machine's broken proc has already been reported (last night I had 5 procs with a 2H wep one right after the other), but I couldn't find a report for Frost Strike. I made a DK of my own and I immediately noticed Frost Strike's huge damage - significantly higher than Heart Strike and Scourge Strike, even when using a 2H wep instead of DW with ToT. At 60lvl my unbuffed DK crits for 1900 Frost Strike, 1700 Heart Strike and 1100 + 400 Scourge Strike. In dungeons I have reached 2200 Heart Strike (with Hysteria) and 2700 Frost Strike. As you can see, these numbers are a bit too high for Frost Strike... Like, VERY high. We're talking about my 60lvl DK doing Hellfire Citadel dungeons, in case you're interested.

People claim that Frost Strike gains improper values from external + bonus % dmg sources (Hellfire Superiority, Luck of the Draw etc.). I hope the community and the testers can do more extensive testing on this issue so that it doesn't happen that Frost completely overshadows Unholy and Blood due to beneficial bugs like this.

Thank you!

**Comment by Lothloryen** [comment]
Hello there,

Thank you very much for your report. I would be very grateful if we could have a more precise or extensive report about Killing machine proc and the bad scaling damage from the frost strike so I can know what is wrong with them.

**Comment by Lothloryen** [comment]
Killing Machine proc was indeed proccing too much, I have corrected it already. Regarding the Frost strike, I would like to have more information and proofs.

Thank you very much.

**Comment by Lothloryen** [comment]
Killing machine proc correction will be live in 15 minutes.

More information regarding the frost strike would be helpfull.

**Comment by AgeAgainst** [comment]
Frost Strike does more than Obliterate does at level 80.  

https://imgur.com/e8Keocf,qJN1k1b#0
https://imgur.com/e8Keocf,qJN1k1b#1

Fixing Killing Machine is a step in the right direction.  But imagine I was using this talent build here:

I did 2 blood strikes in that rotation.  I missed a pestilence and had to redo Icy Touch/Plague Strike.

The problem comes from Scent of Blood in conjunction with Killing Machine proccing too much.  Traditional frost DKs in patch 3.3.5a would use 0/53/18 build.  On Algalon, the common build is 15/56.  Scent of Blood has always been considered a blood tanking talent.

Scent of Blood:  You have a 15% chance after dodging, parrying or taking direct damage to gain the Scent of Blood effect, causing your next (2/3) melee hit(s) to generate 10 runic power.

As per http://wowwiki.wikia.com/Scent_of_Blood
Patch 3.1.0: Will now proc on a dodge, parry or when taking damage, and now grants 10 runic power per charge.  Internal cooldown removed.  This talent can no longer be triggered by environmental damage.

The problem lays with a perceived bug with this talent (unsure how to document it, so if you can show me how to document, I'd be glad to).  It's believed that this talent procs when anyone in your group/raid dodges, parries or takes direct damage.  With no internal cooldown just about every single boss in Ulduar has 10/25 people taking direct damage, many times all at once.  Some frost DKs don't even use Blood Strike in their rotation, saving the global cooldown for Frost Strike.

**Comment by Vaxsysl** [comment]
AND that talent procs on environmental damage, same as fury warrior's enrage and owlkins' Owlkin Frenzy, almost pure 10% increased damage.

**Comment by taralej90** [comment]
So just take a DK with SoB in a group with a tank, both go to the Bombardment in Icecrown, let the tank aggro 10-15 mobs, and see if SoB will proc on the DK. I really doubt that SoB procs from other group members though - it would be outrageous if it did.

**Comment by taralej90** [comment]
The issue I'm addressing here more importantly is Frost Strike. Can anyone do some kind of testing on this one? Frost Strike really shouldn't be dealing that much damage...

**Comment by AgeAgainst** [comment]
I would like to retract my previous statements about FS hitting harder than Obliterate.  I have spoken with another DK with how recount shows FS and Oblit.

In my picture above, Frost Strike does more damage than Obliterate.  However, I failed to take into account one thing and was reminded of it.  When Killing Machine procs to make Frost Strike crit, both weapons crit.  Obliterate can crit with the main hand and normal hit with offhand, thus showing Obliterate to appear to do less damage.  Additionally, Obliterate is a physical damage, whereas Frost Strike's damage is frost.  

I was using the Highlord's Nemesis Trainer in Ebon Hold.  For the purposes of hit rating, this is considered a boss and requires 8% hit rating.  I have nothing to to prove or disprove that it either a) takes into account a player's armor penetration or b) if it factors in any partial damage due to resists.  I do know however that at one time when I only had 20/20 expertise on this same trainer, I didn't miss or have a single attack dodged, which led me to believe that 20 was the expertise cap on this server.  I quickly found out that this was not the case.  This makes me believe that the trainer in Ebon Hold does not take into account a player's armor penetration which would in turn make Obliterate hit even harder.

I am leaving my images up as a warning to others not to make my mistake in using recount as a testing tool to find out if one ability is hitting harder than the other.  I was in a group and tried Taralej90's suggestion about SoB and I did not see Scent of Blood proc once.  Perhaps not enough testing, but that's best suited for a different thread.

**Comment by AgeAgainst** [comment]
I've looked over my gear and consulted another DK.  I personally do not have the gear to do further testing, so I will leave the testing to the better geared, but I would like to share some thoughts.

While purely anecdotal, I honestly do not believe Frost Strike is doing considerably more damage than other abilities. Taralej90 did not mention which runeforge was being used, but I remember using Cinderglacier leveling up until I could use Fallen Crusader.  Cinderglacier has a chance to increase the damage by 20% for next 2 attacks that deal frost or shadow damage.  If Cinderglacier procced just before Taralej90 got back to back Killing Machine procs, that could easily account for the extra damage. When I did my "test", I was DW Frost, using Razorice MH, FC OH.  The numbers reported in my images reflect 10% extra frost damage.

I also do not have a lot of armor penetration.  I had more armor penetration, hitting a boss level trainer would show Obliterate doing more damage.  If I had Grim Tol, it should show Obliterate doing even more damage.

Again, purely anecdotal.  I am unable to test any further.  I hope another DK with an armor pen trinket and more armor pen than I have continues testing, but I don't know if this is really an issue requiring further testing.

**Comment by Runecell** [comment]
Done several hours of testing as to the scaling of frost strike

What i did -  to make the test as accurate as possible i removed all strength procs, (eg rune of fallen crusader, strength greatness card ) so my dmg wasnt modified

2nd i used a 2handed weapon so i wouldnt get conflicting data from recount, since what recount does is adds the mainhand attack and offhand attack ( for Threat of Thessarian) together and averages them for 2 hits, IE if your MH does 2k and offhand 2mg does 1k recount will show froststrike doing 1,5k over 2 hits, couple that with the fact that killing machine will guarantee a crit with both weapons, a none KM frost strike could crit with 1 weapon and not the other, further skewing the results.

So, the tooltip for froststrike  is - 60-61% weapon dmg + 165.
The modifiers for frost strike are
10% frost dmg - Black Ice
15% dmg - Tundra stalker
20% dmg - Glacier Rot
10% dmg - Blood of the north
10% dmg - razor Ice

These are multiplicative with i think the exception of razor ice, which is an added 10% to the total, so the formula would be - [{(weapon damage x 0.605) + 165)} x 1.669]  x 1.1 

my weapon dmg for the test was  1909-2275 ( average 2092)
so plug it in, u get  (2092 x 0.605) +165 = 1430.66 before modifiers                                           1430.66 x 1.669 = 2387.8  + 10% from razor ice  for a total of 2626.55 average damage

plug in the min/max and u get 

HITS - 2407min - 2852max with a 2627 average, depending on how many decimal places u go to
CRITS - 6089min  7216max with a 6646 average
For the test, i made sure i had 5 stacks of razor ice, and had diseases rolling before starting using frost strike, and i made sure to consume KM procs with howlingblast /icy touch sometimes to get a more even number of hits and crits 

Results were, from 31 crits and 29 hits of frost strike

HITS -  Min - 2477  average - 2664     max - 2880
CRITS- Min 6134   average -6714    max -7287

From the results u can see that they are within 1% of the formula, and i can pass this off as to the number of decimal places, and how these figures are rounded up during the calculation

To sum up,  i cant see any issue with how frost strike is scaling
 60 frost strikes may not big a big enough sample size, so if anyone else wants to do a bigger one feel free

**Comment by taralej90** [comment]
@AgeAgainst - I used Razorice. Frost Strike does not consume Cinderglacier - feel free to check this out, if you want.

Just last night I got completely wrecked in Naxx10 by a FDK with wrong gemming, no enchants and 500 GS lower than mine, who consistently dealt +3.5k DPS. His main damage was Frost Strike, critting for around 7k at 3.4k GS with two KotEB Revered swords. He was one of the lowest geared members of the raid, yet the only person he was not able to out-DPS was the 4.5k GS Fire Mage in the group.

Everyone on this server knows Frost Strike hits harder than it is supposed to, and it is further shown in raids where FDKs out-perform all other classes with similar gear, including other DKs. Frost out-performs everything on this current content. Why?

So far my doubts go for Glacier Rot and Merciless Combat, seeing as these affect Frost Strike, HB and IT (HB and IT also hit too hard, but they're not a big part of Frost's rotation, so nobody notices).

**Comment by Runecell** [comment]
@taralej90 
no, everyone is salty and doesnt take the time to actually think about why frost does so well. 
why?
Its because of 2 things, KIlling machine, and Threat of thessarian

Frost has always had killing machine, which is basically an AUTO CRIT, but before threat of thessarian everyone used 2 handed weapons. so it didnt proc that much.
THERE WAS NO DUELWIELD FROST DURING NAXX AND ULDUAR this spec didnt exist

so, now u introduce a spec that procs an auto crit from auto attacks, at starting level, 
Put simple frost does crazy well at low gear levels because it doesnt rely on decent stat pools just like arcane mages and destro locks.
As gear gets better and other classes are hitting those soft caps it will be a totally different show
Frost doesnt have a gear problem,  u just get steadily better with it
where as say a fury warrior, with shit weapons and low crit will do badly, a fury warrior with 232 weapons and arp cap, will piss on an equally geared frost DK


As a side note, you clearly know nothing about DKs if you thing blood subspec is wrong at low gear levels.
Subversion - 9% crit on obliterate
Bladed armour - 350+ free attack power, which scales with raid buffs and armour potion
dark conviction -5% crit on all abilities

this far out weighs the benifits from unholy subspec at low gear levels, since necrosis is completely reliant on weapon damage

And finally 3.5k dps  with 3.4k gs isnt anything special

**Comment by taralej90** [comment]
@Runecell - I know, I thought of this myself first too. But it still doesn't seem plausible of a reason for FS to hit so hard. As I said, I was doing crazy hits with 2H too, not just DW. I'm not being salty, DK was one of my main classes back on retail, Frost too - I just honestly think that Frost Strike hits for too much.

The DK I mentioned had wrong gems, no enchanta, semi-tank gear, yet he was hitting for the same numbers you mentioned. Sure, raid buffs and all, but are you sure that is all?

Molten had the same bug where Frost Strike was doubledipping into buffs & auras. People claim it's the same here, so thank you for linking target dummy damage, but by default I don't think that's where the problem lies. And I'll be glad if proof and numbers can be brought on the table like above, instead of you calling people clueless on GitHub... Thanks.

**Comment by taralej90** [comment]
And if what you're saying is true, why do FDKs still obliterate the DPS meters even at end gear?

**Comment by StylerTyler** [comment]
I agree with Runecell's statements on this matter.

@taralej90 
how can you possibly tell that 'FDKs still obliterate the DPS meters even at end gear' if we are still at Ulduar gear level? Things will drastically change when other melee classes start obtaining more ArP , which scales exponentially in value while FDK benefits really poorly from it. 
The value of KM will drop because of high natural crit, while Frost Strike damage compared to Obliterate will drop because of natural ArP on gear, Bladed Armor will lose value because the amount of armor on DPS plate will not increase significantly -> talent gets worse because the % of your total AP it provides will get lower with progressing gear levels. 
In addition, higher natural haste/ArP will shift the damage distribution towards Auto attacks/OB, which will favor the Unholy sub HEAVILY.
To sum it up, FDK damage is very high indeed, but not fundamentally bugged or broken. 
In the future, please desist from spreading false informations without solid numerical and experiantial proof.

-Styles

**Comment by taralej90** [comment]
@StylerTyler - I do not have numerical and experiantial proof because I do not have the base to gather this proof from. I am not "spreading" any false information, I simply opened a discussion/ brainstorming/ data feed for what is already a widespread belief ingame. Thank you for the scaling explanation, which is something that I knew, but by "endgame" I was talking about current endgame.

Oh well then, Loth you can close this issue. I couldn't gather any proof on the matter, regardless of all the rumors; however I managed to gather a lot of "Internet heroes" at least.

**Comment by taralej90** [comment]
Unsubscribing from this as well.
